subsys/bluetooth/host/hci_core.c

```diff
@@ -1130,6 +1130,19 @@ static int set_le_scan_enable(u8_t enable)
 }
 #endif /* CONFIG_BT_OBSERVER */
 
+static inline bool rpa_is_new(void)
```

Comment:
Note: Having this function as inline avoids compilation warnings for "defined but not used".
I think this is the best way to handle this conditional struct variables, and we can add more of these for e.g. bt_dev.rl_entries. That should enable us to clean up a lot so that we can use IS_ENABLED much more.